src/app/productConfig/productConfigModal.tpl.html

```diff
@@ -48,14 +48,20 @@ <h4 class="panel-title border-bottom-small" translate>
               <span ng-repeat-end></span>
               <label class="btn btn-default-form u-textLeft custom-amount">
                 <div class="form-group form-group-default u-inline">
-                  <input class="form-control form-control-subtle number" type="text" name="amount"
+                  <input class="form-control form-control-subtle number"
+                         type="number"
+                         name="amount"
                          ng-model="$ctrl.customAmount"
                          ng-change="$ctrl.changeCustomAmount($ctrl.customAmount)"
-                         ng-pattern="/^[1-9]\d*(\.\d+)?$/">
+                         ng-min="1"
+                         ng-max="10000000"
+                         ng-pattern="/^[0-9]+(\.[0-9]{1,2})?$/">
                 </div>
               </label>
               <div role="alert" ng-messages="$ctrl.itemConfigForm.amount.$error" ng-if="($ctrl.itemConfigForm.amount | showErrors)">
-                <div class="help-block" ng-message="pattern" translate>Amount must be a valid amount and at least $1</div>
+                <div class="help-block" ng-message="min" translate>Amount must be at least $1.00</div>
+                <div class="help-block" ng-message="max" translate>Amount must be less than $10,000,000.00</div>
```

Comment:
`Amount cannot be more than $10,000,000.00`? The wording for less than equal is always wierd... ng-max should allow a million flat though right?